go/vt/vttablet/onlineddl/executor.go

```diff
@@ -4003,16 +4003,27 @@ func (e *Executor) updateDDLAction(ctx context.Context, uuid string, actionStr s
 
 func (e *Executor) updateMigrationMessage(ctx context.Context, uuid string, message string) error {
 	log.Infof("updateMigrationMessage: uuid=%s, message=%s", uuid, message)
-	query, err := sqlparser.ParseAndBind(sqlUpdateMessage,
-		sqltypes.StringBindVariable(message),
-		sqltypes.StringBindVariable(uuid),
-	)
-	if err != nil {
+
+	maxlen := 2048
+	update := func(message string) error {
+		message = strings.ToValidUTF8(message, "�")
+		if len(message) > maxlen {
+			message = message[0:maxlen]
+		}
```

Comment:
Since `message` is a `TEXT`, `16383` would be a safe maxlen. But we still then need to flip the order, so first truncate and then convert to valid UTF-8. It could lead to a `�` at the end if then we break the last potentially valid unicode char, but I think that's a totally ok trade off if the message is that long already anyway.

Reply:
done